tango/step.py

```diff
@@ -49,30 +55,31 @@ def get_args(tp):  # type: ignore
 T = TypeVar("T")
 
 
+_random_for_step_names = random.Random()
+
+
 class Step(Registrable, Generic[T]):
     """
-    This class defines one step in your experiment. To write your own step, just derive from this class
+    This class defines one step in your experiment. To write your own step, derive from this class
     and overwrite the :meth:`run()` method. The :meth:`run()` method must have parameters with type hints.
 
     ``Step.__init__()`` takes all the arguments we want to run the step with. They get passed
     to :meth:`run()` (almost) as they are. If the arguments are other instances of ``Step``, those
     will be replaced with the step's results before calling :meth:`run()`. Further, there are four special
     parameters:
 
-    * ``step_name`` contains an optional human-readable name for the step. This name is used for
+    :param step_name: contains an optional human-readable name for the step. This name is used for
       error messages and the like, and has no consequence on the actual computation.
-    * ``cache_results`` specifies whether the results of this step should be cached. If this is
+    :param cache_results: specifies whether the results of this step should be cached. If this is
       ``False``, the step is recomputed every time it is needed. If this is not set at all,
       and :attr:`CACHEABLE` is ``True``, we cache if the step is marked as :attr:`DETERMINISTIC`,
       and we don't cache otherwise.
-    * ``step_format`` gives you a way to override the step's default format (which is given in :attr:`FORMAT`).
-    * ``step_config`` is the original raw part of the experiment config corresponding to this step.
+    :param step_format: gives you a way to override the step's default format (which is given in :attr:`FORMAT`).
+    :param step_config: is the original raw part of the experiment config corresponding to this step.
       This can be accessed via the :attr:`config` property within each step's :meth:`run()` method.
-    * ``step_executor`` is the :class:`~tango.executor.Executor` being used to run the step.
-      This can be accessed via the :attr:`executor` property within each step's :meth:`run()` method.
     """
 
-    DETERMINISTIC: bool = False
+    DETERMINISTIC: bool = True
     """This describes whether this step can be relied upon to produce the same results every time
     when given the same inputs. If this is ``False``, the step can't be cached, and neither can any
     step that depends on it."""
```

Comment:
I don't think this is true. We warn the user but we still allow it.

Reply:
6db2c29be0699ac3041d750b17356f6c14c003be